x-pack/plugins/infra/server/services/rules/rule_data_client.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { once } from 'lodash';
+import { CoreSetup, Logger } from 'src/core/server';
+import { TECHNICAL_COMPONENT_TEMPLATE_NAME } from '../../../../rule_registry/common/assets';
+import { RuleRegistryPluginSetupContract } from '../../../../rule_registry/server';
+import { logThresholdRuleDataNamespace } from '../../../common/alerting/logs/log_threshold';
+import type { InfraFeatureId } from '../../../common/constants';
+import { RuleRegistrationContext, RulesServiceStartDeps } from './types';
+
+export const createRuleDataClient = ({
+  ownerFeatureId,
+  registrationContext,
+  getStartServices,
+  logger,
+  ruleDataService,
+}: {
+  ownerFeatureId: InfraFeatureId;
+  registrationContext: RuleRegistrationContext;
+  getStartServices: CoreSetup<RulesServiceStartDeps>['getStartServices'];
+  logger: Logger;
+  ruleDataService: RuleRegistryPluginSetupContract['ruleDataService'];
+}) => {
+  const initializeRuleDataTemplates = once(async () => {
+    const componentTemplateName = ruleDataService.getFullAssetName(
+      `${registrationContext}-mappings`
+    );
+
+    const indexNamePattern = ruleDataService.getFullAssetName(`${registrationContext}*`);
+
+    if (!ruleDataService.isWriteEnabled()) {
+      return;
+    }
+
+    await ruleDataService.createOrUpdateComponentTemplate({
+      name: componentTemplateName,
+      body: {
+        template: {
+          settings: {
+            number_of_shards: 1,
+          },
+          mappings: {
+            properties: {
+              [logThresholdRuleDataNamespace]: {
+                properties: {
+                  serialized_params: {
+                    type: 'keyword',
```

Comment:
out of curiosity, why is this a keyword?

Reply:
`type: "keyword"` and `index: false` seemed to be the most parsimonious field configuration that doesn't cause a mapping explosion but is retrievable via `fields` (which is what the alerts table search strategy uses). I would have preferred `type: "object"` and `enabled: false`, but then the field can't be retrieved via `fields`.

Reply:
@afgomez Back in this PR Could this be related to the issue you have been discussing with @jasonrhodes #113003 ? 

I didn't completely follow up the discussion you had there, I just though it might be worth mentioning that Felix was indexing params in this PR (https://github.com/elastic/kibana/pull/104341/files?file-filters%5B%5D=.ts#diff-4b27f4f59bd7ebc79225d9e379a663944592f8595e811a156ea75f66dfba679a) that you might want to have a look at.

Reply:
@mgiota thanks! It seems this code has been removed since this PR was merged though